### PR TITLE
[CW-3710] Check all of updated route for selfRouteWithdraw.

### PR DESCRIPTION
**Pull request by kishiguro**

### CODE REVIEW

File: server/zclient.go [hunk 1]
```diff
@@ -654,41 +669,27 @@ func (z *zebraClient) loop() {
...
+							}
 						}
-						var vrf uint16 = 0
 						fmt.Println("dst-len", len(dst), "msg.Vrf", msg.Vrf)
-						for index, p := range dst {
+						for pos, path := range dst {
+							if NlriPrefix(path.GetNlri().String()) == "0.0.0.0/0" {
+								continue
+							}
+							var vrf uint16 = 0
```
Comment (yoichi-cloudwan) [inline server/zclient.go:695]: VRF table ID in the kernel is uint32. Is this OK?

**Review by yoichi-cloudwan — APPROVED** [review]
Just FYI, Kernel VRF table ID is 32 bits. You might want to take care of this in the near future.

File: server/zclient.go [hunk 2]
```diff
@@ -569,6 +569,21 @@ func (z *zebraClient) addGlobalVrfsToWatchEvent(ev WatchEvent) WatchEvent {
...
+	if pstr != "" {
+		nhop := p.GetNexthop()
+		withdraw := ""
+		if p.IsWithdraw {
+			withdraw = "withdraw"
+		}
+		if selfRouteWithdraw {
+			withdraw = "self-route withdraw"
+		}
+		fmt.Printf("vrf %d [%d] %s %s %s\n", vrf, index, pstr, nhop.String(), withdraw)
```
Comment (jkmar) [inline server/zclient.go]: Is there logger available?

File: server/zclient.go [hunk 3]
```diff
@@ -654,41 +669,27 @@ func (z *zebraClient) loop() {
 						"Topic": "Zebra",
 					}).Debug("MultiplePathListLen %d, PathListLen %d", len(msg.MultiPathList), len(msg.PathList))
 					for _, dst := range msg.MultiPathList {
-						if len(dst) == 0 {
-							continue
-						}
-						path := dst[0]
+						// We need to check all of Path for self route withdraw.
```
Comment (jkmar) [inline server/zclient.go]: Is the logic changed in this PR? If so then I would suggest adding some tests to verify it.

File: server/zclient.go (hunk 1 above)
Comment (kishiguro) [reply server/zclient.go:695]: upstream gobgp uses uint16.  maybe we should consider to expand to uint32 in near future.

File: server/zclient.go (hunk 2 above)
Comment (kishiguro) [reply server/zclient.go]: yes, just converted to logger.

File: server/zclient.go (hunk 3 above)
Comment (kishiguro) [reply server/zclient.go]: test has been added.

**Comment by kishiguro** [comment]
merged part of v3.1.1 sync.

**Pull request closed without merging**